openhcl/underhill_core/src/emuplat/tpm.rs

```diff
@@ -2,50 +2,68 @@
 // Licensed under the MIT License.
 
 use guest_emulation_transport::GuestEmulationTransportClient;
+use openhcl_attestation_protocol::igvm_attest::get::runtime_claims::AttestationVmConfig;
+use openhcl_attestation_protocol::igvm_attest::get::AK_CERT_RESPONSE_BUFFER_SIZE;
 use thiserror::Error;
-use tpm::ak_cert::GetAttestationReport;
 use tpm::ak_cert::RequestAkCert;
-use underhill_attestation::AttestationVmConfig;
+use underhill_attestation::AttestationType;
 
 #[allow(missing_docs)] // self-explanatory fields
 #[derive(Debug, Error)]
-pub enum TpmGetAttestationReportError {
+pub enum TpmAttestationError {
     #[error("failed to get a hardware attestation report")]
     GetAttestationReport(#[source] tee_call::Error),
-    #[error("failed to create IgvmAttest AK_CERT request")]
-    CreateIgvmAttestAkCertRequest(#[source] underhill_attestation::IgvmAttestError),
+    #[error("failed to create the IgvmAttest AK_CERT request")]
+    CreateAkCertRequest(#[source] underhill_attestation::IgvmAttestError),
 }
 
-/// An implementation of [`GetAttestationReport`].
-pub struct TpmGetAttestationReportHelper {
-    tee_call: Box<dyn tee_call::TeeCall>,
+/// An implementation of [`RequestAkCert`].
+#[derive(Clone)]
+pub struct TpmRequestAkCertHelper {
+    get_client: GuestEmulationTransportClient,
+    tee_call: Option<Box<dyn tee_call::TeeCall>>,
+    attestation_type: AttestationType,
     attestation_vm_config: AttestationVmConfig,
+    attestation_agent_data: Option<Vec<u8>>,
 }
 
-impl TpmGetAttestationReportHelper {
+impl TpmRequestAkCertHelper {
     pub fn new(
-        tee_call: Box<dyn tee_call::TeeCall>,
+        get_client: GuestEmulationTransportClient,
+        tee_call: Option<Box<dyn tee_call::TeeCall>>,
+        attestation_type: AttestationType,
         attestation_vm_config: AttestationVmConfig,
+        attestation_agent_data: Option<Vec<u8>>,
     ) -> Self {
         Self {
+            get_client,
             tee_call,
+            attestation_type,
             attestation_vm_config,
+            attestation_agent_data,
         }
     }
 }
 
-impl GetAttestationReport for TpmGetAttestationReportHelper {
-    fn get_attestation_report(
+#[async_trait::async_trait]
+impl RequestAkCert for TpmRequestAkCertHelper {
+    fn create_ak_cert_request(
         &self,
         ak_pub_modulus: &[u8],
         ak_pub_exponent: &[u8],
         ek_pub_modulus: &[u8],
         ek_pub_exponent: &[u8],
         guest_input: &[u8],
     ) -> Result<Vec<u8>, Box<dyn std::error::Error + Send + Sync>> {
-        let igvm_attest_request_helper =
+        let tee_type = match self.attestation_type {
+            AttestationType::Snp => Some(tee_call::TeeType::Snp),
+            AttestationType::Tdx => Some(tee_call::TeeType::Tdx),
+            AttestationType::Host => None,
+            AttestationType::VbsUnsupported => panic!("VBS not supported yet"), // TODO VBS
```

Comment:
maybe its better to strip out this variant entirely, and have a single top-level panic! / error message in worker.rs?